pkg/collector/osmlogs_collector.go

```diff
@@ -73,70 +60,53 @@ func (collector *OsmLogsCollector) Collect() error {
 		if err != nil {
 			return err
 		}
-		collectDataFromNamespaces(collector, namespacesInMesh, rootPath, meshName)
-		collectControllerLogs(collector, rootPath, meshName)
-	}
-
-	return nil
-}
-
-// * Collects data for namespaces in a given mesh
-func collectDataFromNamespaces(collector *OsmLogsCollector, namespaces []string, rootPath, meshName string) error {
-	for _, namespace := range namespaces {
-		namespaceMetadataFile := filepath.Join(rootPath, meshName+"_"+namespace+"_"+"metadata")
-		namespaceMetadata, err := utils.RunCommandOnContainer("kubectl", "get", "namespaces", namespace, "-o=jsonpath={..metadata}", "-o", "json")
+		err = collectDataFromNamespaces(collector, namespacesInMesh, rootPath, meshName)
 		if err != nil {
 			return err
 		}
-		err = utils.WriteToFile(namespaceMetadataFile, namespaceMetadata)
+		err = collectControllerLogs(collector, rootPath, meshName)
 		if err != nil {
 			return err
 		}
-		collector.AddToCollectorFiles(namespaceMetadataFile)
+	}
 
-		namespaceServicesFile := filepath.Join(rootPath, meshName+"_"+namespace+"_"+"services")
-		namespaceServices, err := utils.RunCommandOnContainer("kubectl", "get", "services", "-n", namespace)
-		if err != nil {
-			return err
-		}
-		err = utils.WriteToFile(namespaceServicesFile, namespaceServices)
-		if err != nil {
-			return err
-		}
-		collector.AddToCollectorFiles(namespaceServicesFile)
+	return nil
+}
 
-		namespaceServicesAllConfigsFile := filepath.Join(rootPath, meshName+"_"+namespace+"_"+"services"+"_"+"all_configs")
-		namespaceServicesAllConfigs, err := utils.RunCommandOnContainer("kubectl", "get", "services", "-n", namespace, "-o", "json")
-		if err != nil {
-			return err
-		}
-		err = utils.WriteToFile(namespaceServicesAllConfigsFile, namespaceServicesAllConfigs)
-		if err != nil {
-			return err
-		}
-		collector.AddToCollectorFiles(namespaceServicesAllConfigsFile)
+func collectResourceToFile(collector *OsmLogsCollector, rootPath, fileName string, kubeCmds []string) error {
+	resourceFile := filepath.Join(rootPath, fileName)
```

Comment:
This could create resource files that are empty, would it be possible to check if there are that type of resources in the ns then create the file?

Reply:
that is a really good point... open to suggestions on how to do this since this function is sometimes called for cmds that are not `kubectl get` so we can't simply check if the `kubectl get <resource>` output is empty

Reply:
also cannot just check if the `output` variable is empty because, for example, when using `-o json`, empty looks like this:
```
$ k get endpoints -n bookthief -o json
{
    "apiVersion": "v1",
    "items": [],
    "kind": "List",
    "metadata": {
        "resourceVersion": "",
        "selfLink": ""
    }
}
```

Reply:
The current implementation doesn't write empty files. If there are no resources of that type in a namespace, then `No resources found in bookbuyer namespace.` gets written to the table file, while the json file (like what Sanya commented) looks like the one above.

Imho leave it as is. When someone looks at these files, it is immediately apparent to them that there is no resource of that type.

![image](https://user-images.githubusercontent.com/13926417/116309981-4ddc9c00-a75e-11eb-8a52-ad9507403700.png)

Reply:
What if we removed the `-o json` flag when checking if the resource exists? And did something like 
`k get endpoints |  wc -c` ??

Reply:
The `No resources foudn in bookbuyer namespaces` is not being written, instead it generates an empty file. @johnsonshi to address this in a follow up PR